cpp-package/src/board_shim.cpp

```diff
@@ -168,6 +168,22 @@ BrainFlowArray<double, 2> BoardShim::get_board_data ()
     delete[] buf;
     return matrix;
 }
+BrainFlowArray<double, 2> BoardShim::get_board_data (int num_samples)
+{
```

Comment:
need to add a check that num_samples is less or equal to the current size of a buffer(check original implementation of get_board_data method)

Reply:
if num_samples is bigger need to return std::min(num_samples, current_size)

Reply:
Done, hopefully, in the latest commit, I've addressed them. Also, I've added the overloaded function for Java, C++, and Python. Please check and suggest if these are good to go. And then I can work upon other languages.